py3status/modules/gpmdp.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+"""
+Display currently playing song from Google Play Music Desktop Player.
+
+Configuration parameters:
+    - cache_timeout : how often we refresh this module in seconds (5s default)
+
+Requires
+    - gpmdp
+    - gpmdp-remote
+
+@author Aaron Fields https://twitter.com/spirotot
+@license BSD
+"""
+
+# import your useful libs here
+from time import time
+from subprocess import check_output
+import shlex
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+
+    cache_timeout = 5
+    format = '{info}'
+
+    def gpmdp(self, i3s_output_list, i3s_config):
+        result = u'♫ ' + self.format
+
+        if check_output(shlex.split('gpmdp-remote status')).decode('ascii', 'ignore').strip() == 'Paused':
+            result = ''
+        else:
+            cmds = ['info', 'title', 'artist', 'album', 'status', 'current',
+                    'status', 'time_total', 'time_current', 'album_art']
+
+            for cmd in cmds:
+                if str('{' + cmd + '}') in result:
+                    result = result.replace(str('{' + cmd + '}'),check_output(shlex.split('gpmdp-remote %s' % (cmd))).decode('ascii', 'ignore').strip())
```

Comment:
This method minimizes the number of times that `gpmdp-remote` needs to be run, as well as provides a simple 1:1 mapping between format string parameters and the output of gpmdp-remote. It's efficient (in LOC, but it's also not bad in terms of runtime), and easy to use/understand (don't know if  I would say the same for the common `format.format()` convention I keep seeing -- at least name the _variable_/_member_ `format` something like `format_str` to differentiate it from the _method_ `format`? That being said, I'm _not_ saying that I think this GPMDP method is suitable for every module...). 

The alternative dictionary method would be to parse out the artist, album, track, current time, and total track length every time the module is called. That requires 4-5 runs of gpmdp-remote, as well as a bunch of extra code. With this method, if someone wants just the default output ('info'), gpmdp-remote only gets run once. 'artist' and 'track'? Two runs.

`str()` casts work in Python2 and Python3. 

Let me know what you think.